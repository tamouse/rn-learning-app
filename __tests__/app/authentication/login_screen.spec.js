import { shallow, mount } from "enzyme"
import React from "react"
import LoginScreen from "../../../app/authentication/login_screen"

describe('LoginScreen', function () {
  it('renders the LoginScreen', function () {
    const submitLogin = jest.fn()
    const actual = shallow(<LoginScreen loading={false} submitLogin={submitLogin} errorMsg={''}/>)
    expect(actual).toMatchSnapshot()
  });
});