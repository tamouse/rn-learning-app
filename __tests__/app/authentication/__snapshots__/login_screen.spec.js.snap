// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders the LoginScreen 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen
    errorMsg=""
    loading={false}
    submitLogin={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="account slug"
          style={
            Object {
              "borderColor": "#9e9e9e",
              "borderRadius": 4,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
              "width": 250,
            }
          }
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="user name"
          style={
            Object {
              "borderColor": "#9e9e9e",
              "borderRadius": 4,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
              "width": 250,
            }
          }
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="password"
          style={
            Object {
              "borderColor": "#9e9e9e",
              "borderRadius": 4,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
              "width": 250,
            }
          }
          value=""
        />
        <Button
          accessibilityLabel="Login Button"
          buttonType="primary"
          disabled={true}
          onPress={[Function]}
        >
          Log In
        </Button>
        <Component
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              id="loading_modal_text"
              style={
                Object {
                  "backgroundColor": "rebeccapurple",
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "900",
                  "padding": 20,
                }
              }
            >
              I'm Loading, yo
            </Text>
          </View>
        </Component>
      </View>,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "flex": 1,
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="account slug"
            style={
              Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              }
            }
            value=""
          />,
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="user name"
            style={
              Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              }
            }
            value=""
          />,
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="password"
            style={
              Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              }
            }
            value=""
          />,
          <Button
            accessibilityLabel="Login Button"
            buttonType="primary"
            disabled={true}
            onPress={[Function]}
          >
            Log In
          </Button>,
          undefined,
          <Component
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            visible={false}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                id="loading_modal_text"
                style={
                  Object {
                    "backgroundColor": "rebeccapurple",
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "900",
                    "padding": 20,
                  }
                }
              >
                I'm Loading, yo
              </Text>
            </View>
          </Component>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCapitalize": "none",
            "autoCorrect": false,
            "onChangeText": [Function],
            "placeholder": "account slug",
            "style": Object {
              "borderColor": "#9e9e9e",
              "borderRadius": 4,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
              "width": 250,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCapitalize": "none",
            "autoCorrect": false,
            "onChangeText": [Function],
            "placeholder": "user name",
            "style": Object {
              "borderColor": "#9e9e9e",
              "borderRadius": 4,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
              "width": 250,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "autoCapitalize": "none",
            "autoCorrect": false,
            "onChangeText": [Function],
            "placeholder": "password",
            "style": Object {
              "borderColor": "#9e9e9e",
              "borderRadius": 4,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
              "width": 250,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessibilityLabel": "Login Button",
            "buttonType": "primary",
            "children": "Log In",
            "disabled": true,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "Log In",
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                id="loading_modal_text"
                style={
                  Object {
                    "backgroundColor": "rebeccapurple",
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "900",
                    "padding": 20,
                  }
                }
              >
                I'm Loading, yo
              </Text>
            </View>,
            "hardwareAccelerated": false,
            "onRequestClose": [Function],
            "visible": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                id="loading_modal_text"
                style={
                  Object {
                    "backgroundColor": "rebeccapurple",
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "900",
                    "padding": 20,
                  }
                }
              >
                I'm Loading, yo
              </Text>,
              "style": Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "I'm Loading, yo",
                "ellipsizeMode": "tail",
                "id": "loading_modal_text",
                "style": Object {
                  "backgroundColor": "rebeccapurple",
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "900",
                  "padding": 20,
                },
              },
              "ref": null,
              "rendered": "I'm Loading, yo",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="account slug"
            style={
              Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              }
            }
            value=""
          />
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="user name"
            style={
              Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              }
            }
            value=""
          />
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="password"
            style={
              Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              }
            }
            value=""
          />
          <Button
            accessibilityLabel="Login Button"
            buttonType="primary"
            disabled={true}
            onPress={[Function]}
          >
            Log In
          </Button>
          <Component
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            visible={false}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                id="loading_modal_text"
                style={
                  Object {
                    "backgroundColor": "rebeccapurple",
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "900",
                    "padding": 20,
                  }
                }
              >
                I'm Loading, yo
              </Text>
            </View>
          </Component>
        </View>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "flex": 1,
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="account slug"
              style={
                Object {
                  "borderColor": "#9e9e9e",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "margin": 5,
                  "padding": 5,
                  "width": 250,
                }
              }
              value=""
            />,
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="user name"
              style={
                Object {
                  "borderColor": "#9e9e9e",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "margin": 5,
                  "padding": 5,
                  "width": 250,
                }
              }
              value=""
            />,
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="password"
              style={
                Object {
                  "borderColor": "#9e9e9e",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "margin": 5,
                  "padding": 5,
                  "width": 250,
                }
              }
              value=""
            />,
            <Button
              accessibilityLabel="Login Button"
              buttonType="primary"
              disabled={true}
              onPress={[Function]}
            >
              Log In
            </Button>,
            undefined,
            <Component
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              visible={false}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  id="loading_modal_text"
                  style={
                    Object {
                      "backgroundColor": "rebeccapurple",
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "900",
                      "padding": 20,
                    }
                  }
                >
                  I'm Loading, yo
                </Text>
              </View>
            </Component>,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCapitalize": "none",
              "autoCorrect": false,
              "onChangeText": [Function],
              "placeholder": "account slug",
              "style": Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCapitalize": "none",
              "autoCorrect": false,
              "onChangeText": [Function],
              "placeholder": "user name",
              "style": Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": true,
              "autoCapitalize": "none",
              "autoCorrect": false,
              "onChangeText": [Function],
              "placeholder": "password",
              "style": Object {
                "borderColor": "#9e9e9e",
                "borderRadius": 4,
                "borderWidth": 1,
                "margin": 5,
                "padding": 5,
                "width": 250,
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessibilityLabel": "Login Button",
              "buttonType": "primary",
              "children": "Log In",
              "disabled": true,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "Log In",
            "type": [Function],
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  id="loading_modal_text"
                  style={
                    Object {
                      "backgroundColor": "rebeccapurple",
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "900",
                      "padding": 20,
                    }
                  }
                >
                  I'm Loading, yo
                </Text>
              </View>,
              "hardwareAccelerated": false,
              "onRequestClose": [Function],
              "visible": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  id="loading_modal_text"
                  style={
                    Object {
                      "backgroundColor": "rebeccapurple",
                      "color": "white",
                      "fontSize": 20,
                      "fontWeight": "900",
                      "padding": 20,
                    }
                  }
                >
                  I'm Loading, yo
                </Text>,
                "style": Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "width": "100%",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "I'm Loading, yo",
                  "ellipsizeMode": "tail",
                  "id": "loading_modal_text",
                  "style": Object {
                    "backgroundColor": "rebeccapurple",
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "900",
                    "padding": 20,
                  },
                },
                "ref": null,
                "rendered": "I'm Loading, yo",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
